### ISSUE TITLE
Not able to get detekt custom rule to work.

### ISSUE THREAD

**Issue by alpeshvas** [detekt/detekt#7759]
# Issue 
Not able to get detekt custom rule to work. Followed all relevant issues (#6194 #3264 #3897  ) and documentation, still not able to figure out the cause.
Used template from detekt.

## Expected Behavior

On running `./gradlew detekt` it should catch a rule violation for one of the class.
## Observed Behavior

Unit tests are working fine as defined [here](https://github.com/alpeshvas/detekt-poc/blob/main/detekt-common-config/src/test/kotlin/headout/detekt/rules/NoDummyClassRuleTest.kt) 
But when applying the plugin to the root build.gradle file, it seems to not catching violation in this [class](https://github.com/alpeshvas/detekt-poc/blob/main/src/main/kotlin/Dummy.kt) for Dummy rule
## Steps to Reproduce

Live example repository - https://github.com/alpeshvas/detekt-poc

- `./gradlew detekt` on root, it doesn't fail. It should fail
## Context

Trying to write a custom rules for our usecase.

## Your Environment
Set up:
Custom rules are in the `detekt-common-config` module
```
plugins {
    kotlin("jvm") version "2.0.0"
}

apply(plugin = "kotlin")

repositories {
    mavenCentral()
}

dependencies {
    compileOnly("io.gitlab.arturbosch.detekt:detekt-api:1.23.7")
    testImplementation("io.gitlab.arturbosch.detekt:detekt-test:1.20.0")
    testImplementation("org.jetbrains.kotlin:kotlin-test:1.8.0")
    testImplementation(platform("org.junit:junit-bom:5.10.0"))
    testImplementation("org.junit.jupiter:junit-jupiter")
}
tasks.test {
    useJUnitPlatform()
}
```

Applied in root project with this setup:
Using `detektPlugins(project(":detekt-common-config"))`
```
import org.gradle.kotlin.dsl.test
import org.gradle.kotlin.dsl.testImplementation

plugins {
    id("java")
    kotlin("jvm") version "2.0.0"
    id("io.gitlab.arturbosch.detekt") version "1.23.7"
}

group = "com.detekt.poc"
version = "1.0-SNAPSHOT"

repositories {
    mavenCentral()
}

dependencies {
    detektPlugins(project(":detekt-common-config"))
    testImplementation(platform("org.junit:junit-bom:5.10.0"))
    testImplementation("org.junit.jupiter:junit-jupiter")
}

tasks.test {
    useJUnitPlatform()
}
```
* Version of detekt used: 1.23.7
* Version of Gradle used (if applicable): Gradle 8.10.2
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS 
* Link to your project (if it's a public repository):
https://github.com/alpeshvas/detekt-poc

**Comment by alpeshvas** [comment]
It started working after I enabled allRules flag to true throught `detekt` dsl, this is so weird. There is no such thing mentioned in the documentation, adding custom rule in the generated detekt.yml config file should work right?

**Comment by BraisGabin** [comment]
Yes, it should. I'm not sure what's going on.

What happens when you put the config on your project configuration directly?

**Comment by alpeshvas** [comment]
> Yes, it should. I'm not sure what's going on.
> 
Can you check it out in the public repository I shared? Tried different things, no luck.

**Comment by cortinico** [comment]
I briefly looked into your reproducer @alpeshvas and I'm not able to find the root cause of your problem. At first I thought it was related to you using the root project as sample project, but moving the sample code to a separate `sample` module hasn't solved.

It seems like the rule is not executed altogether. The rule set is however loaded by detekt.

**Comment by alpeshvas** [comment]
> I briefly looked into your reproducer @alpeshvas and I'm not able to find the root cause of your problem. At first I thought it was related to you using the root project as sample project, but moving the sample code to a separate `sample` module hasn't solved.
> 
> 
> 
> It seems like the rule is not executed altogether. The rule set is however loaded by detekt.

Let me know if you need anything else from my side but reproducer has everything to reproduce the issue.

**Comment by alpeshvas** [comment]
@cortinico @BraisGabin following up ^

**Comment by BraisGabin** [comment]
I don't have right now the time to debug the problem. You can checkout https://github.com/BraisGabin/detekt-junit-rules this repo (or create a new one from the template) and use jitpack using the commit version to find if it works. And then keep it changing it until it stop working. This way you can find exactly what's wrong on detekt or on your configuration.

What seems clear is that there is something missing in detekt because it's not easy to debug this issues. So if you find something please keep posting here so we know which is the problem and we find a way to spot those type of issues easier.

**Comment by alpeshvas** [comment]
> I don't have right now the time to debug the problem. You can checkout https://github.com/BraisGabin/detekt-compiler-rules this repo (or create a new one from the template) and use jitpack using the commit version to find if it works. And then keep it changing it until it stop working. This way you can find exactly what's wrong on detekt or on your configuration.
> 
> 
> 
> What seems clear is that there is something missing in detekt because it's not easy to debug this issues. So if you find something please keep posting here so we know which is the problem and we find a way to spot those type of issues easier.

Tried, couldn't figure out the issue
